cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -1069,12 +1113,17 @@ public:
 
             for (int channel = 0; channel < NUM_ACTIVE_CHANNELS; ++channel)
             {
-                privatized_decode_op[channel].Init(num_output_levels[channel],
-                                                   upper_level[channel],
-                                                   lower_level[channel]);
+              cudaError_t error = privatized_decode_op[channel].Init(num_output_levels[channel],
```

Comment:
Refactored this part to only return an error when _not_ querying temporary storage requirements.

Reply:
Thanks for clarifying. I think what we want is to always return the error in case of an overflow but also make sure to return a proper value for `temp_storage_bytes`

Reply:
Changed to return `temp_storage_bytes` of `1` when invoked for a scenario that would overflow 👍

Reply:
...and also added tests to verify we initialize `temp_storage_bytes` 🙂